### ISSUE TITLE
Handle rate limiting from RetryLink

### ISSUE THREAD

**Issue by mxstbr** [apollographql/apollo-link#995]
**Expected Behavior**

When a server returns a 429 rate limited response with a `Retry-After` header, the RetryLink does not retry until after the specified delay.

**Actual Behavior**

The RetryLink keeps retrying.

**Comment by JoviDeCroock** [comment]
Thanks for the issue report, I'll take a look at it tonight.

**Comment by mxstbr** [comment]
I tried that, unfortunately, the `error` does not contain the network status code.

**Comment by JoviDeCroock** [comment]
Aha, this is indeed an issue we've been facing frequently. I'm going to try and look into this but it's quite hard to reproduce at this point. That's valuable information though thank you for that.

**Comment by JoviDeCroock** [comment]
This feels like weird behavior to me:

https://codesandbox.io/s/ry49xr9nzm

Look at the logs, the statusCode seems to be contained in the networkError.

**Comment by JoviDeCroock** [comment]
@mxstbr I have updated the sandbox to show the statusCode inside the retryLink aswell. If this does not fix your issue I can dig deeper into it but I'll need some more information.

**Comment by mxstbr** [comment]
Yes exactly, I only want to retry after the server allows me to retry again.

**Comment by JoviDeCroock** [comment]
Sorry for the confusion, I have been hunting for that statusCode for a while that it was drawing my attention of the main issue.

**Comment by JoviDeCroock** [comment]
Hey @mxstbr if you would find the time could you check if this works for you. https://github.com/apollographql/apollo-link/pull/1004

Sorry about getting back to you this late

**Comment by helfer** [comment]
@mxstbr I agree that the default retry link should check the `Retry-After` header. For now unfortunately there doesn't seem to be a way to work around it, because the delay function doesn't get the error (which should contain the status code btw, so if it doesn't, that's another issue).
